src/code/z_camera.c

```diff
@@ -1,3070 +1,6778 @@
-    0x00, 0x00, 0x42, 0x70, 0x00, 0x00, 0x3F, 0x80, 0x00, 0x00, 0xBF, 0x80, 0x00, 0x00, 0xBF, 0x80, 0x00, 0x00,
-    0xBF, 0x80, 0x00, 0x00, 0xBF, 0x80, 0x00, 0x00, 0xBF, 0x80, 0x00, 0x00, 0xBF, 0x80, 0x00, 0x00,
-};
-
-u8 D_8011D9F4[] = {
-    0x8F, 0xFF, 0x05, 0x04, 0x00, 0x14, 0x00, 0x02, 0x42, 0x70, 0x00, 0x00, 0x3F, 0x80, 0x00, 0x00, 0x00, 0x00,
-    0x00, 0x00, 0x40, 0xA0, 0x00, 0x00, 0x42, 0x48, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x41, 0xA0, 0x00, 0x00,
-    0x43, 0x96, 0x00, 0x00, 0x82, 0xFF, 0x21, 0x21, 0x00, 0x05, 0x00, 0x02, 0x42, 0x70, 0x00, 0x00, 0x3F, 0x80,
-    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x41, 0x20, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-    0x41, 0xA0, 0x00, 0x00, 0xC3, 0x16, 0x00, 0x00, 0x12, 0xFF, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x42, 0x70,
-    0x00, 0x00, 0x3F, 0x80, 0x00, 0x00, 0xBF, 0x80, 0x00, 0x00, 0xBF, 0x80, 0x00, 0x00, 0xBF, 0x80, 0x00, 0x00,
-    0xBF, 0x80, 0x00, 0x00, 0xBF, 0x80, 0x00, 0x00, 0xBF, 0x80, 0x00, 0x00,
-};
-
-u8 D_8011DA6C[] = {
-    0x45, 0x3E, 0xA0, 0x00, 0x44, 0x2F, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x44, 0xDB, 0x60, 0x00,
-    0x45, 0x55, 0x70, 0x00, 0xC3, 0xBE, 0x00, 0x00, 0xC5, 0x43, 0x00, 0x00, 0x45, 0x45, 0x80, 0x00,
-    0x43, 0x75, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xC1, 0x20, 0x00, 0x00, 0x43, 0x70, 0x00, 0x00,
-};
-
-u8 D_8011DA9C[] = {
-    0x45, 0x45, 0x80, 0x00, 0x45, 0x06, 0x60, 0x00, 0x00, 0x00, 0x00, 0x00, 0x44, 0xBD, 0x60, 0x00,
-    0x45, 0x81, 0x10, 0x00, 0xC4, 0x50, 0xC0, 0x00, 0xC5, 0x3E, 0x00, 0x00, 0x45, 0x81, 0x38, 0x00,
-    0x43, 0x66, 0x00, 0x00, 0xC2, 0x48, 0x00, 0x00, 0x44, 0x16, 0x00, 0x00, 0xC2, 0x96, 0x00, 0x00,
-};
-
-u8 D_8011DACC[] = {
-    0x44, 0xC4, 0x40, 0x00, 0x45, 0x66, 0x00, 0x00, 0x45, 0x67, 0x40, 0x00, 0x43, 0xC5, 0x80, 0x00,
-};
-
-u8 D_8011DADC[] = {
-    0x43, 0xA0, 0x00, 0x00, 0x43, 0xA0, 0x00, 0x00, 0x43, 0xA0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-};
-
-u8 D_8011DAEC[] = {
-    0xF8, 0x30, 0xFC, 0x18, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-};
-
-s16 D_8011DAFC[] = {
-    1, 2, 58, 3, 4, 62,
-};
-
-s16 D_8011DB08 = 1008;
-
-s16 D_8011DB0C = 342;
-
-s32 D_8011DB10 = 0;
-
-s32 D_8011DB14 = 0;
-
-// BSS
-extern GlobalContext* D_8015BD7C;
-extern DbgCamera D_8015BD80;
-extern Vec3f D_8015BD84;
-extern Vec3f D_8015BD90;
-extern Vec3f D_8015BD9C;
-extern f32 D_8015BDC8;
-extern Vec3s D_8015CE46;
-extern CollisionPoly* D_8015CE4C;
+#define RELOAD_PARAMS \
+    (camera->animState == 0 || camera->animState == 0xA || camera->animState == 0x14 || R_RELOAD_CAM_PARAMS)
 
-/*===============================================================*/
-typedef struct {
-    Vec3f unk_00;
-    Vec3f unk_0C;
-    CollisionPoly* unk_18;
-    char unk_1C[0x8];
-    s32 unk_24;
-} struct_80043D18;
+#define PCT(x) ((x)*0.01f)
+#define NEXTSETTING ((values++)->val)
+#define NEXTPCT PCT(NEXTSETTING)
+
+#define ONEPNTDEM ((Unique9OnePointDemo*)camera->paramData)
 
```

Comment:
feel like we could prob afford a more explicit name even if its a bit longer. I have to try real hard to make out what ONEPNTDEM means everytime i see it